drivers/bluetooth/hci/hci_spi_st.c

```diff
@@ -114,6 +113,43 @@ struct bt_hci_ext_evt_hdr {
 	uint16_t len;
 } __packed;
 
+int bluenrg_bt_spi_reset(bool updater_mode)
+{
+	int err = 0;
+	/* Assert reset */
+	if (!updater_mode) {
+		gpio_pin_set_dt(&rst_gpio, 1);
+		k_sleep(K_MSEC(DT_INST_PROP_OR(0, reset_assert_duration_ms, 0)));
+		gpio_pin_set_dt(&rst_gpio, 0);
+	} else {
+#if DT_HAS_COMPAT_STATUS_OKAY(st_hci_spi_v2)
+		return -ENOTSUP;
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_hci_spi_v1)
+		gpio_pin_set_dt(&rst_gpio, 1);
+		gpio_pin_interrupt_configure_dt(&irq_gpio, GPIO_INT_DISABLE);
+		/* Configure IRQ pin as output and force it high */
+		err = gpio_pin_configure_dt(&irq_gpio, GPIO_OUTPUT_ACTIVE);
+		if (err) {
+			return err;
+		}
+		/* Add reset delay and release reset */
+		k_sleep(K_MSEC(DT_INST_PROP_OR(0, reset_assert_duration_ms, 0)));
+		gpio_pin_set_dt(&rst_gpio, 0);
+		/* Give firmware some time to read the IRQ high */
+		k_sleep(K_MSEC(5));
+		gpio_pin_interrupt_configure_dt(&irq_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+		/* Reconfigure IRQ pin as input */
+		err = gpio_pin_configure_dt(&irq_gpio, GPIO_INPUT);
+		if (err) {
+			return err;
+		}
+		/* Emulate possibly missed rising edge IRQ by signaling the IRQ semaphore */
+		k_sem_give(&sem_request);
+#endif /* DT_HAS_COMPAT_STATUS_OKAY(st_hci_spi_v2) */
```

Comment:
Do you want to handle a `#else` case to trigger an error ?

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/638a47f804deb7b76f996ce2c85c49b5b5e4a8dd/drivers/bluetooth/hci/CMakeLists.txt#L22-L28
According to this CMakeLists, `hci_spi_st.c` only compiles when there is a compatibility for either `st_hci_spi_v1` or `st_hci_spi_v2`, so I think `#else` isn't necessary. What do you think?

Reply:
Ok. We generally use if/else in that case (using comment to clarify the else branch /* DT_HAS_COMPAT_STATUS_OKAY(st_hci_spi_v2) */).
The advantage I see (the only one I guess) is that it makes it clear to the reader that no case was omitted.
Not blocking though.

Reply:
Updated. Done